cpp/CPP14.g4

```diff
@@ -21,2345 +22,2013 @@
-;
-
-Operator
-:
-	'operator'
-;
-
-Override
-:
-	'override'
-;
-
-Private
-:
-	'private'
-;
-
-Protected
-:
-	'protected'
-;
-
-Public
-:
-	'public'
-;
-
-Register
-:
-	'register'
-;
-
-Reinterpret_cast
-:
-	'reinterpret_cast'
-;
-
-Return
-:
-	'return'
-;
-
-Short
-:
-	'short'
-;
-
-Signed
-:
-	'signed'
-;
-
-Sizeof
-:
-	'sizeof'
-;
-
-Static
-:
-	'static'
-;
-
-Static_assert
-:
-	'static_assert'
-;
-
-Static_cast
-:
-	'static_cast'
-;
-
-Struct
-:
-	'struct'
-;
-
-Switch
-:
-	'switch'
-;
-
-Template
-:
-	'template'
-;
-
-This
-:
-	'this'
-;
-
-Thread_local
-:
-	'thread_local'
-;
-
-Throw
-:
-	'throw'
-;
-
-True
-:
-	'true'
-;
-
-Try
-:
-	'try'
-;
-
-Typedef
-:
-	'typedef'
-;
-
-Typeid
-:
-	'typeid'
-;
-
-Typename
-:
-	'typename'
-;
-
-Union
-:
-	'union'
-;
-
-Unsigned
-:
-	'unsigned'
-;
-
-Using
-:
-	'using'
-;
-
-Virtual
-:
-	'virtual'
-;
-
-Void
-:
-	'void'
-;
-
-Volatile
-:
-	'volatile'
-;
-
-Wchar
-:
-	'wchar_t'
-;
-
-While
-:
-	'while'
-;
-
-/*Operators*/
-LeftParen
-:
-	'('
-;
-
-RightParen
-:
-	')'
-;
-
-LeftBracket
-:
-	'['
-;
-
-RightBracket
-:
-	']'
-;
-
-LeftBrace
-:
-	'{'
-;
-
-RightBrace
-:
-	'}'
-;
-
-Plus
-:
-	'+'
-;
-
-Minus
-:
-	'-'
-;
-
-Star
-:
-	'*'
-;
-
-Div
-:
-	'/'
-;
-
-Mod
-:
-	'%'
-;
-
-Caret
-:
-	'^'
-;
-
-And
-:
-	'&'
-;
-
-Or
-:
-	'|'
-;
-
-Tilde
-:
-	'~'
-;
-
-Not
-:
-	'!'
-;
-
-Assign
-:
-	'='
-;
-
-Less
-:
-	'<'
-;
-
-Greater
-:
-	'>'
-;
-
-PlusAssign
-:
-	'+='
-;
-
-MinusAssign
-:
-	'-='
-;
-
-StarAssign
-:
-	'*='
-;
-
-DivAssign
-:
-	'/='
-;
-
-ModAssign
-:
-	'%='
-;
-
-XorAssign
-:
-	'^='
-;
-
-AndAssign
-:
-	'&='
-;
-
-OrAssign
-:
-	'|='
-;
-
-LeftShift
-:
-	'<<'
-;
-
-rightShift
-:
-//'>>'
-	Greater Greater
-;
-
-LeftShiftAssign
-:
-	'<<='
-;
-
-rightShiftAssign
-:
-//'>>='
-	Greater Greater Assign
-;
-
-Equal
-:
-	'=='
-;
-
-NotEqual
-:
-	'!='
-;
-
-LessEqual
-:
-	'<='
-;
-
-GreaterEqual
-:
-	'>='
-;
-
-AndAnd
-:
-	'&&'
-;
-
-OrOr
-:
-	'||'
-;
-
-PlusPlus
-:
-	'++'
-;
-
-MinusMinus
-:
-	'--'
-;
-
-Comma
-:
-	','
-;
-
-ArrowStar
-:
-	'->*'
-;
-
-Arrow
-:
-	'->'
-;
-
-Question
-:
-	'?'
-;
-
-Colon
-:
-	':'
-;
-
-Doublecolon
-:
-	'::'
-;
-
-Semi
-:
-	';'
-;
-
-Dot
-:
-	'.'
-;
-
-DotStar
-:
-	'.*'
-;
-
-Ellipsis
-:
-	'...'
-;
-
-theoperator
-:
-	New
-	| Delete
-	| New '[' ']'
-	| Delete '[' ']'
-	| '+'
-	| '-'
-	| '*'
-	| '/'
-	| '%'
-	| '^'
-	| '&'
-	| '|'
-	| '~'
-	| '!'
-	| '='
-	| '<'
-	| '>'
-	| '+='
-	| '-='
-	| '*='
-	| '/='
-	| '%='
-	| '^='
-	| '&='
-	| '|='
-	| '<<'
-	| rightShift
-	| rightShiftAssign
-	| '<<='
-	| '=='
-	| '!='
-	| '<='
-	| '>='
-	| '&&'
-	| '||'
-	| '++'
-	| '--'
-	| ','
-	| '->*'
-	| '->'
-	| '(' ')'
-	| '[' ']'
-;
-
-/*Lexer*/
-fragment
-Hexquad
-:
-	HEXADECIMALDIGIT HEXADECIMALDIGIT HEXADECIMALDIGIT HEXADECIMALDIGIT
-;
-
-fragment
-Universalcharactername
-:
-	'\\u' Hexquad
-	| '\\U' Hexquad Hexquad
-;
-
-Identifier
-:
-/*
+   : Assign val = Octalliteral{if($val.text.compareTo("0")!=0) throw new InputMismatchException(this);};
+   
+   /*Derived classes*/
+   baseclause
```

Comment:
Why do you use whitespaces before rule declarations? It does not make sense.

Reply:
Antlr4Formatter does it.   I have no objection at all to you making a change to Antlr4Formatter.